src/Spice86.Core/Emulator/OperatingSystem/DosPathResolver.cs

```diff
@@ -185,11 +171,11 @@ private string GetFullDosPathIncludingRoot(string absoluteOrRelativeDosPath) {
             if (StartsWithDosDriveAndVolumeSeparator(dosPath)) {
                 length = 3;
             }
-            return (_driveMap[_currentDrive].MountedHostDirectory, dosPath[length..]);
+            return (_dosDriveManager.CurrentDrive.MountedHostDirectory, dosPath[length..]);
         } else if (StartsWithDosDriveAndVolumeSeparator(dosPath)) {
```

Comment:
Small logic error here maybe (not introduce by this PR):
IsPathRooted returns true if StartsWithDosDriveAndVolumeSeparator is true, so I guess this else if is never reached and we always resolve to current drive?

Reply:
Yeah probably.

But this code makes me head hurt.

Is unit tested and always works, but I dare not touch it. :D